app/models/comment.rb

```diff
@@ -202,9 +204,16 @@ def wrap_timestamps_if_video_present!
   def shorten_urls!
     doc = Nokogiri::HTML.fragment(processed_html)
     doc.css("a").each do |anchor|
-      unless anchor.to_s.include?("<img") || anchor.to_s.include?("<del") || anchor.attr("class")&.include?("ltag")
-        anchor.content = strip_url(anchor.content) unless anchor.to_s.include?("<img") # rubocop:disable Style/SoleNestedConditional
+      anchor_inner_html = anchor.inner_html
+      next if anchor_inner_html.include?("<img")
+
+      anchor_content = anchor.content
+      urls = anchor_content.scan(URI_REGEXP).flatten.compact
+      urls.each do |url|
+        anchor_content.sub!(/#{Regexp.escape(url)}/, strip_url(url))
       end
+      anchor_inner_html.sub!(/#{Regexp.escape(anchor.content)}/, anchor_content)
+      anchor.inner_html = anchor_inner_html
```

Comment:
This confused me a little bit. Do we really need to update this temporary variable? Wouldn't it be enough to assign the new `anchor_content` directly on the next line?

```suggestion
      anchor.inner_html = anchor_content
```

In that case I think we wouldn't need the `anchor_inner_html` local variable at all and can upgrade the guard clause above to the following:

```ruby
next if anchor.inner_html.include?("<img")
```

Reply:
I wasn't able to replace attribute substrings directly using `sub!` . Hence I had to use temporary variable assignments to do that instead. I'll admit it does look convoluted, I'll see if it can be simplified. The `anchor_inner_html` is not needed in the first line though, I'll take that out.

Reply:
`anchor.inner_html` and `anchor.content` don't interact correctly, and replacing sub strings in `anchor.content` directly causes both to update incorrectly. This problem is avoided with the temporary variable assignment to `anchor_content`. I was however able to get rid of `anchor_inner_html`.